numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -3423,26 +3431,54 @@ PyArray_Where(PyObject *condition, PyObject *x, PyObject *y)
         /* Get the result from the iterator object array */
         ret = (PyObject*)NpyIter_GetOperandArray(iter)[0];
 
-        NPY_BEGIN_THREADS_NDITER(iter);
+        PyArray_Descr **dts = NpyIter_GetDescrArray(iter);
+        PyArray_Descr *dtx = dts[2];
+        PyArray_Descr *dty = dts[3];
+        npy_intp itemsize = dts[0]->elsize;
```

Comment:
```suggestion
        npy_intp itemsize = common_dt->elsize;
```
Use `common_dt` everywhere.  The iterator enforces that after all (if it didn't we would actually have to do the cast below.  To be fair, when we don't use that "trivial copy" fast-path, that could be faster even, but that is a different thing than this PR).